### ISSUE TITLE
W0611: Unused import enum (unused-import) when checking a file called signal.py

### ISSUE THREAD

**Issue by hitimr** [pylint-dev/pylint#9207]
### Bug description

pylint detects a false positive warning (W0611) when checking a file that is named "signal.py". Even if the file is empty.

### Command used

```shell
touch signal.py && pylint signal.py
```

### Pylint output

```python
************* Module signal
signal.py:2:0: W0611: Unused import enum (unused-import)
```

### Expected behavior

No warning generated (when parsing an empty file)

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
```

### OS / Environment

Kubuntu 22.04 LTS

**Comment by PMExtra** [comment]
Also recurs in `http.py`
